runtime/drivers/sqlite/registry.go

```diff
@@ -242,6 +246,7 @@ func (c *connection) EditInstance(_ context.Context, inst *drivers.Instance) err
 			olap_connector = $2,
 			repo_connector = $3,
 			admin_connector = $4,
+			catalog_connector = $5,
 			updated_on = $5,
```

Comment:
Numbers are off – doesn't this breaks `EditInstance`?

Reply:
Weirdly `sqlite` doesn't throw an error when doing `update` but throws error on `select` after doing this operation. Updated the unit tests to directly use select.
I also happen to notice that while doing `EditInstance` we are updating the cache with the passed instance object and not with object fetched from db : https://github.com/rilldata/rill/blob/2f7e89c2bff10e647cfb967feb92fec6a837cf69/runtime/registry.go#L278
This doesn't seem correct to me even though everything in instance right now is updated from the object only but we should select the updated object and then update the cache for following reasons:
1. DB can have defaults which may not be captured in the passed instance object. 
2. For this particular case it is possible that cache and db can end up not being in sync in case on concurrent edits.

Reply:
Sounds very weird. Yeah, seems like there's a race condition there (though the `EditInstance` DB call updates all fields, so chance is quite low). Ideally I guess `EditInstance` would return the new one, or alternatively, `FindInstance` should be called under the lock. Will you add an issue for it?

Reply:
Yeah calling a `FindInstance` under the lock should be enough ?